src/coreComponents/mesh/generators/NewGhosting.cpp

```diff
@@ -497,30 +572,38 @@ struct Adjacency
   std::vector< std::vector< TriScalarInt > > values;
 };
 
+// Function to populate the adjacency struct defined directly above
+// Note that we really build I+A, as in the end we will essentially be doing graph traversal by multiplying by (I+A)^n, ((I+A)^T)^n
+// Including I gives you everything in the graph 'up to' n steps away, without it you only get things 'exactly' n steps away
 Adjacency buildAdjacency( MeshGraph const & owned,
                           MeshGraph const & present,
                           FindGeometricalType const & convert )
 {
   Adjacency adjacency;
 
+  // Num matrix indices owned by the current rank, and the num indices corresponding to the 'present' data
   std::size_t const numOwned = std::size( owned.n2pos ) + std::size( owned.e2n ) + std::size( owned.f2e ) + std::size( owned.c2f );
   std::size_t const numOther = std::size( present.n2pos ) + std::size( present.e2n ) + std::size( present.f2e );
 
-  // Aliases
+  // Aliases for the data in adjacency
   std::vector< TriGlbIdx > & ownedNodesIdcs = adjacency.ownedNodesIdcs;
   std::vector< TriGlbIdx > & ownedGlbIdcs = adjacency.ownedGlbIdcs;
   std::vector< TriGlbIdx > & otherGlbIdcs = adjacency.otherGlbIdcs;
   Teuchos::Array< std::size_t > & numEntriesPerRow = adjacency.numEntriesPerRow;
   std::vector< std::vector< TriGlbIdx > > & indices = adjacency.indices;
   std::vector< std::vector< TriScalarInt > > & values = adjacency.values;
 
+  // we can go ahead and set sizes for the current rank matrix data
   ownedNodesIdcs.reserve( std::size( owned.n2pos ) );
   ownedGlbIdcs.reserve( numOwned );
   otherGlbIdcs.reserve( numOther );
   numEntriesPerRow.reserve( numOwned );
   indices.reserve( numOwned );
   values.reserve( numOwned );
 
+  // fill the matrix index data for the 'present' gemetrical entities (nodes, edges, faces)
```

Comment:
Maybe I could rename `otherGlbIdcs` into `presentGlbIdcs` to maintain symmetry with the graphs. Then you can rename at your convenience?

Reply:
I've renamed.